src/util/path.js

```diff
@@ -605,11 +612,22 @@
       coordsStr = currentPath.slice(1).trim();
       coords.length = 0;
 
-      while ((match = re.exec(coordsStr))) {
-        coords.push(match[0]);
-      }
+      var command = currentPath.charAt(0);
+      coordsParsed = [command];
 
-      coordsParsed = [currentPath.charAt(0)];
+      if (command.toLowerCase() === 'a') {
+        // arcs have special flags that apparently don't require spaces so handle special
+        for (var args; (args = regArcArgumentSequence.exec(coordsStr));) {
+          for (var j = 1; j < args.length; j++) {
+            coords.push(args[j]);
+          }
+        }
+      }
```

Comment:
can be this written like:
```js
if (command.toLowerCase() === 'a') {
  // arcs have special flags that apparently don't require spaces so handle special
  coords = regArcArgumentSequence.exec(coordsStr);
}
else {
  coords = [];
  while ((match = re.exec(coordsStr))) {
     coords.push(match[0]);
  }
}
```

would it work?

Reply:
line 613
https://github.com/fabricjs/fabric.js/pull/6571/files#diff-b4a17ab08e7f28e3bf83badb9236d01cR613
could go away

and so the pre declaration here would be just 
https://github.com/fabricjs/fabric.js/pull/6571/files#diff-b4a17ab08e7f28e3bf83badb9236d01cR589
```js
var coords,
```

thre is no point to reuse the same array this is not hot code to optimize.

Reply:
This failed unit tests when I tried it. An arc can have multiple strings of numbers behind it so there needs to be a loop. Also, the exec returns an array that has [0] with a list of matches and then [1..n] with the matches (hence why j starts at 1). Maybe it can be reworked but I'm not sure there's a gain there.
Are you saying you don't like coords.length = 0 and would rather have coords = [] at 613? I didn't care for the former.